File: tkpdstream/src/main/java/com/tokopedia/tkpdstream/chatroom/view/viewmodel/chatroom/GeneratedMessageViewModel.java
package com.tokopedia.tkpdstream.chatroom.view.viewmodel.chatroom;

import com.tokopedia.abstraction.base.view.adapter.Visitable;
import com.tokopedia.tkpdstream.chatroom.view.adapter.chatroom.typefactory.GroupChatTypeFactory;

/**
 * @author by nisie on 3/29/18.
 */

public class GeneratedMessageViewModel extends BaseChatViewModel implements
        Visitable<GroupChatTypeFactory> {

    public static final String TYPE = "generated_msg";

    public GeneratedMessageViewModel(String message, long createdAt, long updatedAt,
                                     String messageId, String senderId, String senderName,
                                     String senderIconUrl, boolean isInfluencer, boolean isAdministrator) {
        super(message, createdAt, updatedAt, messageId, senderId, senderName, senderIconUrl, isInfluencer, isAdministrator);
    }

    @Override
    public int type(GroupChatTypeFactory typeFactory) {
        return typeFactory.type(this);
    }
}

File: features/user/groupchat/src/main/java/com/tokopedia/groupchat/chatroom/kotlin/view/viewmodel/chatroom/GeneratedMessageViewModel.kt
package com.tokopedia.groupchat.chatroom.kotlin.view.viewmodel.chatroom

import com.tokopedia.abstraction.base.view.adapter.Visitable
import com.tokopedia.groupchat.chatroom.kotlin.view.adapter.chatroom.typefactory.GroupChatTypeFactory

/**
 * @author by nisie on 3/29/18.
 */

class GeneratedMessageViewModel(message: String, createdAt: Long, updatedAt: Long,
                                messageId: String, senderId: String, senderName: String,
                                senderIconUrl: String, isInfluencer: Boolean, isAdministrator: Boolean) : BaseChatViewModel(message, createdAt, updatedAt, messageId, senderId, senderName, senderIconUrl, isInfluencer, isAdministrator), Visitable<GroupChatTypeFactory> {

    override fun type(typeFactory: GroupChatTypeFactory): Int {
        return typeFactory.type(this)
    }

    companion object {

        val TYPE = "generated_msg"
    }
}
